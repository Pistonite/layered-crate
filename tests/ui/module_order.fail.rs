use layered_crate::layers;

#[layers]
mod src {
    pub extern crate y;
    #[depends_on(y)]
    pub extern crate z;
}

fn main() {
    z::zzz();
    y::do_something();
}
