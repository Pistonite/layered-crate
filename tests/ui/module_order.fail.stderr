error: Module `z` should be declared before its dependency `y` to ensure top-down readability
 --> tests/ui/module_order.fail.rs:7:22
  |
7 |     pub extern crate z;
  |                      ^
